src/math/trig.c

```diff
@@ -9,568 +9,115 @@
-	1996388621,
-	1998806828,
-	2001206221,
-	2003586778,
-	2005948477,
-	2008291295,
-	2010615209,
-	2012920200,
-	2015206244,
-	2017473320,
-	2019721407,
-	2021950483,
-	2024160528,
-	2026351521,
-	2028523441,
-	2030676268,
-	2032809981,
-	2034924561,
-	2037019987,
-	2039096240,
-	2041153301,
-	2043191149,
-	2045209766,
-	2047209132,
-	2049189230,
-	2051150040,
-	2053091543,
-	2055013722,
-	2056916559,
-	2058800035,
-	2060664132,
-	2062508835,
-	2064334123,
-	2066139982,
-	2067926393,
-	2069693341,
-	2071440807,
-	2073168776,
-	2074877232,
-	2076566159,
-	2078235539,
-	2079885359,
-	2081515602,
-	2083126253,
-	2084717297,
-	2086288719,
-	2087840504,
-	2089372637,
-	2090885104,
-	2092377891,
-	2093850984,
-	2095304369,
-	2096738031,
-	2098151959,
-	2099546138,
-	2100920555,
-	2102275198,
-	2103610053,
-	2104925108,
-	2106220351,
-	2107495769,
-	2108751351,
-	2109987084,
-	2111202958,
-	2112398959,
-	2113575079,
-	2114731304,
-	2115867625,
-	2116984030,
-	2118080510,
-	2119157053,
-	2120213650,
-	2121250291,
-	2122266966,
-	2123263665,
-	2124240379,
-	2125197099,
-	2126133816,
-	2127050521,
-	2127947205,
-	2128823861,
-	2129680479,
-	2130517051,
-	2131333571,
-	2132130029,
-	2132906419,
-	2133662733,
-	2134398965,
-	2135115106,
-	2135811152,
-	2136487094,
-	2137142926,
-	2137778643,
-	2138394239,
-	2138989707,
-	2139565042,
-	2140120239,
-	2140655292,
-	2141170196,
-	2141664947,
-	2142139540,
-	2142593970,
-	2143028233,
-	2143442325,
-	2143836243,
-	2144209981,
-	2144563538,
-	2144896909,
-	2145210091,
-	2145503082,
-	2145775879,
-	2146028479,
-	2146260880,
-	2146473079,
-	2146665075,
-	2146836865,
-	2146988449,
-	2147119824,
-	2147230990,
-	2147321945,
-	2147392689,
-	2147443221,
-	2147473541,
-	2147483647
-};
+/* Compute fixed point cordicsine with table lookup and interpolation
+ *	The cordic sine algorithm converges, when the angle is in the range
+ *	[-pi/2, pi/2).If an angle is outside of this range, then a multiple of
+ *	pi/2 is added or subtracted from the angle until it is within the range
+ *	[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
+ *	has range in [-1.0 to 1.0]
+ *	+------------------+-----------------+--------+--------+
+ *	| thRadFxp	   | cdcsinth	     |thRadFxp|cdcsinth|
+ *	+----+-----+-------+----+----+-------+--------+--------+
+ *	|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ *	+----+-----+-------+----+----+-------+--------+--------+
+ *	| 32 | 28  |  1	   | 32	| 31 |	 1   | 4.28   | 1.31   |
+ *	+------------------+-----------------+--------+--------+
+ */
 
-/* Sine lookup table read */
-static inline int32_t sine_lookup(int idx)
+int32_t sin_fixed(int32_t th_rad_fxp)
 {
-	int32_t s;
-	int i1;
+	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31
+	 *CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31-1))))
+	 *Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain
+	 */
+	static const int32_t cordicsine_lookup[31] = { 843314857, 497837829, 263043837, 133525159,
```

Comment:
i just realized this is `1 << idx` can we update this to use the `BIT` macro? E.g.
```
arr = {
  BIT(31),
  BIT(30),
  ...
  BIT(1),
  BIT(0)
};
```

Reply:
This is not a bit table (2^N) instead its a lookuptable  having 0 to 31 bits. There is a description/comment available in the code with formula above variable cordicsine_lookup[31].

Reply:
got it thanks